pybossa/cache/users.py

```diff
@@ -348,10 +348,21 @@ def delete_user_pref_metadata(name):
 
 @memoize(timeout=ONE_DAY)
 def get_user_preferences(user_id):
-    assert user_id is not None or user_id > 0
-    user_pref = User.query.get(user_id).user_pref or {}
+    user_pref = get_user_pref(user_id)
     return get_user_pref_db_clause(user_pref)
 
+@memoize(timeout=ONE_DAY)
```

Comment:
Enhance get_user_pref_db_clause to create the new db_clause based on user preferences since this get_user_preferences is also used this functions n_available_tasks_for_user (helpers.py)